modules/utxo-lib/src/bitgo/UtxoPsbt.ts

```diff
@@ -68,26 +68,27 @@ export class UtxoPsbt<Tx extends UtxoTransaction<bigint>> extends Psbt {
     return psbt;
   }
 
-  static async fromTransactionComplete(
-    transaction: UtxoTransaction<bigint>,
-    prevOutputs: TxOutput<bigint>[],
-    fetchTransactions: (txids: string[]) => Promise<Record<string, Buffer>>
-  ): Promise<UtxoPsbt<UtxoTransaction<bigint>>> {
-    const psbt = this.fromTransaction(transaction, prevOutputs);
-
-    const txidToIndex: Record<string, number> = {};
-    psbt.data.inputs.forEach((input, index) => {
-      if (!input.tapLeafScript && !input.witnessScript) {
-        txidToIndex[reverseBuffer(transaction.ins[index].hash).toString('hex')] = index;
+  async addNonWitnessUtxos(fetchTransactions: (txids: string[]) => Promise<Record<string, Buffer>>): Promise<this> {
+    const txidToIndex = new Map<string, number[]>();
+    this.data.inputs.forEach((input, index) => {
+      if (!input.witnessUtxo) {
+        throw new Error('Must have witness UTXO for all inputs');
+      }
+      if (!isSegwit(input.witnessUtxo.script, input.redeemScript)) {
+        // Copy, then reverse buffer
+        const txid = reverseBuffer(Buffer.from(this.tx.ins[index].hash)).toString('hex');
+        const indices = txidToIndex.get(txid) || [];
+        indices.push(index);
+        txidToIndex.set(txid, indices);
       }
     });
 
-    const txHexs = await fetchTransactions(Object.keys(txidToIndex));
-    Object.entries(txidToIndex).forEach(([txid, index]) => {
-      psbt.updateInput(index, { nonWitnessUtxo: txHexs[txid] });
+    const txHexs = await fetchTransactions([...txidToIndex.keys()]);
```

Comment:
Since the values are actually `Buffer` not `string`, the name `txHexs` is a misnomer (`txBytes` maybe ?)